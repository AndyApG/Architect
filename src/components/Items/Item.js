import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';
import Handle from './Handle';
import DeleteButton from './DeleteButton';

const Item = ({ children, handleDelete, className, sortable }) => (
  <div className={cx('items-item', className)}>
    { sortable &&
      <div className="items-item__control items-item__control--left">
        <Handle />
      </div>
    }
    <div className="items-item__content">
      {children}
    </div>
    <div className="items-item__control items-item__control--right">
      <DeleteButton onClick={handleDelete} />
    </div>
  </div>
);

Item.propTypes = {
  className: PropTypes.string,
  handleDelete: PropTypes.func.isRequired,
  children: PropTypes.node,
  sortable: PropTypes.bool,
};

Item.defaultProps = {
  className: null,
  children: null,
  sortable: true,
};

export { Item };

export default Item;
