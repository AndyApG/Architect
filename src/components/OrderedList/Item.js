import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';
import Handle from './Handle';
import DeleteButton from './DeleteButton';

const Item = ({ children, handleDelete, className, sortable, ...rest }) => (
  <div className={cx('list-item', className)} {...rest} >
    { sortable &&
      <div className="list-item__control list-item__control--left">
        <Handle />
      </div>
    }
    <div className="list-item__content">
      {children}
    </div>
    <div className="list-item__control list-item__control--right">
      <DeleteButton onClick={handleDelete} />
    </div>
  </div>
);

Item.propTypes = {
  className: PropTypes.string,
  handleDelete: PropTypes.func.isRequired,
  children: PropTypes.node,
  sortable: PropTypes.bool,
};

Item.defaultProps = {
  className: null,
  children: null,
  sortable: true,
};

export { Item };

export default Item;
