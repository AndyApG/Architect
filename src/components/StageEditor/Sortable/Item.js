import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';
import Handle from './Handle';
import DeleteButton from './DeleteButton';

const Item = ({ children, handleDelete, className }) => (
  <div className={cx('stage-editor-sortable-item', className)}>
    <div className="stage-editor-sortable-item__control stage-editor-sortable-item__control--left">
      <Handle />
    </div>
    <div className="stage-editor-sortable-item__content">
      {children}
    </div>
    <div className="stage-editor-sortable-item__control stage-editor-sortable-item__control--right">
      <DeleteButton onClick={handleDelete} />
    </div>
  </div>
);

Item.propTypes = {
  className: PropTypes.string,
  handleDelete: PropTypes.func.isRequired,
  children: PropTypes.node,
};

Item.defaultProps = {
  className: null,
  children: null,
};

export { Item };

export default Item;
