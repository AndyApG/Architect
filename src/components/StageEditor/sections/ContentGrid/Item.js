import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import cx from 'classnames';
import ItemPreview from './ItemPreview';
import ItemChooser from './ItemChooser';
import ItemEditor from './ItemEditor';
import { sizes } from './sizes';

class Item extends Component {
  static propTypes = {
    fieldId: PropTypes.string.isRequired,
    editing: PropTypes.string,
    onToggleItemEdit: PropTypes.func.isRequired,
    handleDelete: PropTypes.func.isRequired,
    onChooseItemType: PropTypes.func.isRequired,
    item: PropTypes.shape({
      content: PropTypes.string,
      type: PropTypes.string,
      size: PropTypes.oneOf([sizes.SMALL, sizes.MEDIUM, sizes.LARGE]),
    }),
    spareCapacity: PropTypes.number,
  };

  static defaultProps = {
    editing: false,
    item: {
      content: null,
      type: null,
      size: null,
    },
    spareCapacity: 0,
  };

  get item() {
    return this.props.item;
  }

  get isNew() {
    return !this.item.type;
  }

  get isEditing() {
    return this.item.id === this.props.editing;
  }

  handleChooseItemType = type =>
    this.props.onChooseItemType(this.props.fieldId, type);

  handleToggleItemEdit = () => {
    this.props.onToggleItemEdit(this.item.id);
  }

  render() {
    const {
      fieldId,
      handleDelete,
      spareCapacity,
    } = this.props;

    const variableClasses = cx(
      'content-grid-item',
      { 'content-grid-item--edit': this.isEditing },
      { 'content-grid-item--new': this.isNew },
    );

    return (
      <div
        className={variableClasses}
        onClick={this.handleToggleItemEdit}
      >
        { !this.isNew &&
          <div className="content-grid-item__preview">
            <ItemPreview
              content={this.item.content}
              type={this.item.type}
              size={this.item.size}
              onDeleteItem={handleDelete}
            />
          </div>
        }

        <div className="content-grid-item__edit">
          <ItemChooser
            show={this.isNew}
            onChooseItemType={this.handleChooseItemType}
          />
          <ItemEditor
            show={!this.isNew}
            name={fieldId}
            type={this.item.type}
            size={this.item.size}
            spareCapacity={spareCapacity}
            onComplete={this.handleToggleItemEdit}
          />
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state, { fieldId, form }) => ({
  item: form.getValues(state, `${fieldId}`),
});

export { Item };

export default connect(mapStateToProps)(Item);
