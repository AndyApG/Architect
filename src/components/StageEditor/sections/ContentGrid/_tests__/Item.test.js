/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { Item } from '../Item';

const mockProps = {
  fieldId: 'foo',
  onToggleItemEdit: () => {},
  handleDelete: () => {},
  onChooseItemType: () => {},
};

describe('<Item />', () => {
  it('can render', () => {
    const component = shallow(<Item {...mockProps} />);

    expect(component).toMatchSnapshot();
  });

  it('when item.type is empty show chooser', () => {
    const component = shallow(<Item {...mockProps} item={{ type: null }} />);
    const componentWithoutChooser = shallow(<Item {...mockProps} item={{ type: 'text' }} />);

    expect(component.find('ItemChooser').prop('show')).toBe(true);
    expect(componentWithoutChooser.find('ItemChooser').prop('show')).toBe(false);
  });

  it('shows edit/preview correctly', () => {
    const componentWithType = shallow(<Item {...mockProps} item={{ type: 'foo' }} />);
    const componentWithEditor = shallow(<Item {...mockProps} editing="foo" item={{ id: 'foo' }} />);
    const componentWithPreview = shallow(<Item {...mockProps} editing={null} item={{ id: 'foo' }} />);

    expect(componentWithType.find('ItemPreview').exists()).toBe(true);
    expect(componentWithType.find('ItemEditor').prop('show')).toBe(true);

    expect(componentWithEditor.hasClass('content-grid-item--edit')).toBe(true);
    expect(componentWithPreview.hasClass('content-grid-item--edit')).toBe(false);
  });
});
