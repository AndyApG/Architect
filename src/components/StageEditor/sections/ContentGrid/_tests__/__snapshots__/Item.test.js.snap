// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> can render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Item
    editing={null}
    fieldId="foo"
    handleDelete={[Function]}
    item={
      Object {
        "content": null,
        "size": null,
        "type": null,
      }
    }
    onChooseItemType={[Function]}
    onToggleItemEdit={[Function]}
    spareCapacity={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <div
          className="content-grid-item__edit"
        >
          <ItemChooser
            onChooseItemType={[Function]}
            show={true}
          />
          <ItemEditor
            content={null}
            name="foo"
            onComplete={[Function]}
            show={false}
            size={null}
            spareCapacity={0}
            type={null}
          />
        </div>,
      ],
      "className": "content-grid-item content-grid-item--new",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ItemChooser
              onChooseItemType={[Function]}
              show={true}
            />,
            <ItemEditor
              content={null}
              name="foo"
              onComplete={[Function]}
              show={false}
              size={null}
              spareCapacity={0}
              type={null}
            />,
          ],
          "className": "content-grid-item__edit",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onChooseItemType": [Function],
              "show": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "content": null,
              "name": "foo",
              "onComplete": [Function],
              "show": false,
              "size": null,
              "spareCapacity": 0,
              "type": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <div
            className="content-grid-item__edit"
          >
            <ItemChooser
              onChooseItemType={[Function]}
              show={true}
            />
            <ItemEditor
              content={null}
              name="foo"
              onComplete={[Function]}
              show={false}
              size={null}
              spareCapacity={0}
              type={null}
            />
          </div>,
        ],
        "className": "content-grid-item content-grid-item--new",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ItemChooser
                onChooseItemType={[Function]}
                show={true}
              />,
              <ItemEditor
                content={null}
                name="foo"
                onComplete={[Function]}
                show={false}
                size={null}
                spareCapacity={0}
                type={null}
              />,
            ],
            "className": "content-grid-item__edit",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onChooseItemType": [Function],
                "show": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "content": null,
                "name": "foo",
                "onComplete": [Function],
                "show": false,
                "size": null,
                "spareCapacity": 0,
                "type": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
